qiskit/pulse/transforms/pulse_instruction_types.py

```diff
@@ -0,0 +1,236 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""
+Special data types.
+"""
+
+from enum import Enum
+from typing import NamedTuple, Union, List, Optional, NewType, Dict, Any, Tuple
+
+import numpy as np
+from qiskit import pulse
+
+
+class PhaseFreqTuple(NamedTuple):
```

Comment:
You can update these named tuples with dataclass since it's more readable (when I wrote the original code I was not able to use dataclass since we needed to support python3.6).